### ISSUE TITLE
如何引用在半监督语义分割

### ISSUE THREAD

**Issue by Aawen123** [zifuwanggg/JDTLosses#2]
作者您好，感谢您提供的优秀工作，我希望将该损失使用在我所做的半监督语义分割中，我的论文中目前所使用的交叉熵损失，请问我按照README中直接引用JDTLoss(from losses.jdt_loss import JDTLoss)即可吗？是否有其他需要注意的情况？

另外在论文中您提到和交叉熵最好的组合是0.75，0.25，请问这一部分在代码的哪里体现？期待您的回答，谢谢！

**Comment by zifuwanggg** [comment]
Thank you for your interest in our work.

Ideally, you only need to replace:
```
loss = cross_entropy_loss(logits, label)
```
with something like:
```
loss = 0.25 * cross_entropy_loss(logits, label) + 0.75 * jdt_loss(logits, label)
```

However, there are several important considerations when using segmentation losses in general. For more details, please refer to the [FAQs](https://github.com/zifuwanggg/JDTLosses/blob/master/FAQs.md).

For the loss weight, please refer to [here](https://github.com/zifuwanggg/JDTLosses/blob/68ce8ac431a0ea07d325dacb3d94965634285362/models/segmentor.py#L182) as well as the [configs](https://github.com/zifuwanggg/JDTLosses/blob/68ce8ac431a0ea07d325dacb3d94965634285362/configs/losses).

**Comment by Aawen123** [comment]
非常感谢您的回复，我尝试把该函数JDTLoss导入到我的代码中，并按照您的建议修改了原有的交叉熵损失，但效果下降明显，我猜测我的操作有问题。README中的意思是只需要把JDTLoss导入到我的代码中就可以了吗？其他的文件是否需要导入？希望您解答，非常感谢！    （另外我在半监督语义分割的训练中使用的是两个GPU。）

**Comment by zifuwanggg** [comment]
你原来是只有交叉熵损失没有其他的损失对吧？你每个GPU上能装几张照片？

半监督经常会把几个loss叠在一起，一般是这样：
```
loss1 = CE(logits1, label1)
loss2 = CE(logits2, label2)
loss3 = CE(logits3, label3)
loss = weight1 * loss1 + weight2 * loss2 + weight3 * loss3
```

你只需要把JDTLoss导入，变成下面这样：
```
loss1 = 0.25 * CE(logits1, label1) + 0.75 * JDT(logits1, label1)
loss2 = 0.25 * CE(logits2, label2) + 0.75 * JDT(logits2, label2)
loss3 = 0.25 * CE(logits3, label3) + 0.75 * JDT(logits3, label3)
loss = weight1 * loss1 + weight2 * loss2 + weight3 * loss3
```

如果效果没有提升的话，我建议最需要调的参数应该是epochs，建议看一下[FAQs](https://github.com/zifuwanggg/JDTLosses/blob/master/FAQs.md)。

**Comment by Aawen123** [comment]
epochs已改为原有的一半，没有其他损失，单个GPU batchsize是2。

非常感谢您的解答，我会继续尝试。

**Comment by zifuwanggg** [comment]
I’m closing this issue due to inactivity. If you still need assistance or have additional information to provide, please feel free to reopen the issue or create a new one.
